catalogue/webapp/services/prismic/index.ts

```diff
@@ -0,0 +1,153 @@
+});
+
+// TODO: move Prismic fetch function out to common and use it here and in server-data/prismic
+// To authenticate a request to the Prismic API graphql endpoint you need a Prismic-ref value that refreshes every 30 seconds.
+// The ref can be used in a 'stale' state for 5 minutes before it expires
+// https://community.prismic.io/t/for-how-long-is-it-safe-to-cache-a-prismic-api-ref/5962
+export const prismicRefFetch = (
+  url: string,
+  options?: Record<string, string>
+): Promise<Response> => {
+  return fetch(url, { ...options, agent: agentKeepAlive });
+};
+
+// Prismic API graphql endpoint expects query requests via GET method
+// This means we need to be able to pass a body in the request, something that is not possible with fetch
+// As a workaround I have used axios to make the request
+export const prismicFetch = (
+  options?: AxiosRequestConfig
+): Promise<AxiosResponse> => {
+  return axios({ ...options });
+};
+
+export async function prismicQuery(
+  endpoint: string,
+  { query, pageSize }: PrismicQueryProps
+): Promise<StoryResultsList<Story> | PrismicApiError> {
+  const graphQuery = `query {
+  allArticless(fulltext: "${query}" sortBy: title_ASC first: ${pageSize}) {
+  edges {
+      node {
+        title
+        _meta { id }
+        contributors {
+          contributor {
+            ...on People {
+              name
+            }
+          }
+        }
+        body {
+          ...on ArticlesBodyStandfirst {
+            primary {
+              text
+            }
+          }
+        }
+        promo {
+          ...on ArticlesPromoEditorialimage {
+            primary {
+              image
+              link
+              caption
+            }
+          }
+        }
+      }
+    }
+  }
+}`;
+  const url = `https://wellcomecollection.prismic.io/graphql?query=${graphQuery}`;
+  const fetchRefUrl = 'https://wellcomecollection.prismic.io/api/v2';
+
+  const headers = {
+    Authorization: `Bearer ${process.env.PRISMIC_BEARER_TOKEN}`,
+    repository: 'wellcomecollection',
+  };
+
+  const prismicRef = await prismicRefFetch(fetchRefUrl, headers);
+  const { refs } = await prismicRef.json();
+  const { ref } = refs[0];
+  headers['Prismic-ref'] = ref;
+  const updatedHeaders = {
+    Authorization: `Bearer ${process.env.PRISMIC_TOKEN}`,
+    repository: 'wellcomecollection',
+    'Prismic-ref': ref,
+  };
+  const options: AxiosRequestConfig = {
+    method: 'GET',
+    url: url,
+    headers: updatedHeaders,
+    data: {
+      query: { graphQuery },
+    },
+  };
+
+  try {
+    const res = await prismicFetch(options);
+    const json = await res.data;
+    const { data } = json;
+    const { allArticless } = data;
+    const stories = await transformStories(allArticless);
+
+    return {
+      type: 'ResultList',
+      results: stories,
+      totalResults: stories.length,
+    };
+  } catch (error) {
+    return prismicApiError();
```

Comment:
Not sure how we handle errors here yet, but would we want to pass `error.message` in as the `description`?